### PR TITLE
Issue/respond to spam events

### PR DESCRIPTION
**Pull request by thatside-zaraffa**
Implemented automatic notifications for adminUser when receiving spam complaints

### CODE REVIEW

File: DependencyInjection/AzineMailgunWebhooksExtension.php [hunk 1]
```diff
@@ -30,6 +34,18 @@ public function load(array $configs, ContainerBuilder $container)
 
         if(array_key_exists(self::PUBLIC_API_KEY, $config))
             $container->setParameter(self::PREFIX."_".self::PUBLIC_API_KEY, $config[self::PUBLIC_API_KEY]);
+        
+        if(array_key_exists(self::TICKET_ID, $config))
```
Comment (azine) [inline DependencyInjection/AzineMailgunWebhooksExtension.php]: As all new config-keys have default values the $config will always have the corresponding key (and value) defined => these "if"-statements are unnecessary.

File: Resources/views/Email/notification.html.twig [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<h3>Spam complaint received</h3>
```
Comment (azine) [inline Resources/views/Email/notification.html.twig]: use message-keys and the "trans"-twig filter everywhere please.

File: Services/AzineMailgunMailerService.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        $this->mailerUser = $mailerUser;
+        $this->ticketId = $ticketId;
+        $this->ticketSubject = $ticketSubject;
+        $this->ticketMessage = $ticketMessage;
+        $this->adminUserEmail = $adminUserEmail;
+    }
+
+    public function  sendSpamComplaintNotification($eventId)
+    {
+        if (is_null($this->validateEmail($this->adminUserEmail))) {
```
Comment (azine) [inline Services/AzineMailgunMailerService.php]: move this ($this->validateEmail($this->adminUserEmail)" before the "if" and assign it to a variable, so you don't have to call it twice.

File: Services/AzineMailgunMailerService.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    public function  sendSpamComplaintNotification($eventId)
+    {
+        if (is_null($this->validateEmail($this->adminUserEmail))) {            
+            /** @var \Swift_Message $message */
+            $message = $this->mailer->createMessage();
+            $message->setTo($this->adminUserEmail)
+                    ->setFrom($this->mailerUser)
+                    ->setSubject('Spam complaint received')
```
Comment (azine) [inline Services/AzineMailgunMailerService.php]: please use a message-key here and translate it

**Comment by azine** [comment]
Hi Bogdan,

thanks for the update. See my comments for required changes.

Do you see a possiblity to also add a functional or unit-test, to test the new functionality?

Cheers & have a nice weekend,
Dominik

**Comment by azine** [comment]
@alister : thanks for your thoughts on this. Currently not setting the "admin_email" disables sending an email. But I think we'll make it a bit more explicit.
PS: is your eventlistener that pushes a message to slack open-source? where can it be found?

**Comment by alister** [comment]
I've not previously open sourced it, as it is somewhat coupled to another internal Slack-class. It's not too complicated though - just a MailgunWebhookSubscriber which makes some strings based on the event. 

Having taken a little while to do some more cleanups in it, I've put a copy at: https://gist.github.com/alister/c63afdaa3f2de7ffd50be34a315c9579

**Comment by azine** [comment]
👍😀

------------------------------

Azine IT Services AG <https://azine.me>
St. Jakobstrasse 60
6330 Cham

Dominik Businger, Partner
dominik.businger@azine-it.ch
Skype: <dominik.businger@azine-it.ch>dominikbusinger
<https://azine.me/skype-call-dominikbusinger>
Office: +41 41 508 57 52 <0041415085752>
Mobile: +41 79 352 93 25 <0041793529325>

2017-07-31 14:06 GMT+02:00 Alister Bulman <notifications@github.com>:

**Pull request closed without merging**